### PR TITLE
Fixed a few combat bugs and changed the religions founded label

### PR DESCRIPTION
**Pull request by xlenstra**
Fixes include:
- Fix `Strength \<in [Friendly Land] tiles\>` not working due to the relevant civ not being passed to `tile.matchesFilter`. This fixes Hemeji castle, Nationalsm & some belief not working.
- Fix the temporary autocracy bonus to apply & groundwork for when I'll turn the 'for [amount] turns' into a conditional
- Renamed the 'religions founded' label to 'religions to be founded', as that is the stat it is actually displaying

### CODE REVIEW

File: core/src/com/unciv/logic/civilization/PolicyManager.kt [hunk 1]
```diff
@@ -47,8 +47,11 @@ class PolicyManager {
     }
 
     fun addPolicyToTransients(policy: Policy) {
-        for (unique in policy.uniqueObjects)
-            policyUniques.addUnique(unique)
+        for (unique in policy.uniqueObjects) {
+            // Should be replaced with a conditional of the same form later
+            if (!unique.text.contains(Regex("for \\[[0-9]*\\] turns")))
```
Comment (yairm210) [inline core/src/com/unciv/logic/civilization/PolicyManager.kt:52]: Please make this a static const

Comment (yairm210) [reply core/src/com/unciv/logic/civilization/PolicyManager.kt:52]: The regex that is
Making the regex every time we add a policy is very inefficient

Comment (SomeTroglodyte) [reply core/src/com/unciv/logic/civilization/PolicyManager.kt:52]: True, Regex objects _can_ and in any case with reuse _should_ have a compiled form of the pattern.
As for the pattern itself - I would recommend standardizing on a "transport" form throughout a project, meaning its presentation to the compiler. I'm slightly in favour of using const val with a raw string everywhere, despite kotlin's raw strings being severely broken. const val would be to enable debugger inspection to make sure the pattern actually contains exactly what's intended.
Lastly, that pattern matches ""for [] turns", probably slightly unintended. `?` = 0 or 1, `+` = 1..∞, `*` = 0..∞, `{n}` = exactly n, `{n,m}` = n..m repetitions...

**Comment by xlenstra** [comment]
Failed checks are in master, I'll wait a few mins before closing & reopening

**Comment by SomeTroglodyte** [comment]
You know, I suspect close/reopen isn't the only/cleanest/fastest way to provoke a fresh check run. There's a re-run jobs button on the Checks tab... Any one of us ever tried that?

**Comment by xlenstra** [comment]
> You know, I suspect close/reopen isn't the only/cleanest/fastest way to provoke a fresh check run. There's a re-run jobs button on the Checks tab... Any one of us ever tried that?

Huh, I never noticed that button before

**Pull request merged**